src/v0/destinations/pinterest_tag/utils.js

```diff
@@ -30,6 +30,56 @@ const ecomEventMaps = [
 
 const USER_NON_ARRAY_PROPERTIES = ['client_user_agent', 'client_ip_address'];
 
+const getHashedValue = (key, value) => {
+  if (Array.isArray(value) && !USER_NON_ARRAY_PROPERTIES.includes(key)) {
+    switch (key) {
+      case 'em':
+      case 'ct':
+      case 'st':
+      case 'country':
+      case 'ln':
+      case 'fn':
+      case 'ge':
+        value = value.map((val) => val.toString().toLowerCase());
+        break;
+      case 'ph':
+      case 'zp':
+        // zip fields should only contain digits
+        value = value.map((val) => val.toString().replace(/\D/g, ''));
+        if (key === 'ph') {
+          // phone numbers should not contain leading zeros
+          value = value.map((val) => val.replace(/^0+/, ''));
+        }
+        break;
+      default:
+    }
+    value = value.map((val) => sha256(val));
+    return [...value];
+  }
+  switch (key) {
+    case 'em':
+    case 'ct':
+    case 'st':
+    case 'country':
+    case 'ln':
+    case 'fn':
+    case 'ge':
+      value = value.toString().toLowerCase();
+      break;
+    case 'ph':
+    case 'zp':
+      // zip fields should only contain digits
+      value = value.toString().replace(/\D/g, '');
+      if (key === 'ph') {
```

Comment:
handle `ph` within `ph` condition itself - `value = value.replace(/^0+/, '');`
for `zp`, handle within single statement - `value.toString().replace(/\D/g, '').replace(/^0+/, '')`

Reply:
Idea of switch itself is to handle individual key separately - again adding `if` condition within switch is meaningless

Reply:
Updated